code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -81,7 +81,8 @@
 		EscapeConfinement()
 
 	if(AICanContinue(possible_targets))
-		DestroySurroundings()
+		if(!QDELETED(target) && !targets_from.Adjacent(target))
```

Comment:
and, since we know that the target both exists and isn't adjacent to us, we know that it'll be a direction and is thus safe to jam into the list without worrying about it.